engine/engine.go

```diff
@@ -1,74 +1,78 @@
 package engine
 
 import (
-	"fmt"
-	"sync"
+	"strings"
+	"time"
+
+	"github.com/longhorn/longhorn-engine/controller/client"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 
 	"github.com/longhorn/longhorn-engine-launcher/rpc"
+	"github.com/longhorn/longhorn-engine-launcher/util"
 )
 
 type Engine struct {
-	lock *sync.RWMutex
-
-	Name       string
-	VolumeName string
+	EngineName string
```

Comment:
Do we need this EngineName? Previously controller has an ID but not name. And I saw the reference of Engine Name below is in fact volume name?

Reply:
We need to generate engine process name somewhere. I think it's better to generate it outside of engine manager so that we can use that name to verify the engine creation.

We will use Engine controller to enable the frontend of the corresponding volume. Hence the name of rpc calls are VolumeXXXX.